scipy/signal/_signaltools.py

```diff
@@ -2452,6 +2456,269 @@ def hilbert2(x, N=None):
+    Examples
+    --------
+    The following plot illustrates the envelope of a signal with variable frequency and
+    a low-frequency drift. To separate the drift from the envelope, a 4 Hz highpass
+    filter is used. The low-pass residuum of the input bandpass filter is utilized to
+    determine an asymmetric upper and lower bound to enclose the signal. Due to the
+    smoothness of the resulting envelope, it is down-sampled from 1000 to 40 samples.
+    Note that the instantaneous amplitude `x_a` and the computed envelope `x_env` are
+    not perfectly identical. This due to the signal not being perfectly periodic as
+    well as the existence of some spectral overlapping of `x_carrier` and `x_drift`.
+    Hence, they cannot be completely separated by a bandpass filter.
+
+    >>> import matplotlib.pyplot as plt
+    >>> import numpy as np
+    >>> from scipy.signal.windows import gaussian
+    >>> from scipy.signal import envelope
+    ...
+    >>> n, T, n_out = 1000, 1/1000, 40
+    >>> t = np.arange(n) * T
+    >>> x_a = gaussian(len(t), 0.2/T)  # instantaneous amplitude
+    >>> x_phi = 60*np.pi*t + 40*np.cos(np.pi*t)  # instantaneous phase
+    >>> x_carrier = x_a * np.cos(x_phi)
+    >>> x_drift = 0.3 * gaussian(len(t), 0.2/T)  # drift
+    ...
+    >>> x_env, x_res = envelope(x_carrier + x_drift, (4, None), n_out=n_out)
+    >>> t_out = np.arange(n_out) * (n / n_out) * T
+    ...
+    >>> fg0, ax0 = plt.subplots(1, 1, tight_layout=True)
+    >>> ax0.set_title(r"$4\,$Hz Highpass Envelope of Drifting Signal")
+    >>> ax0.set(xlabel="Time in seconds", xlim=(0, 1), ylabel="Amplitude")
+    >>> ax0.plot(t, x_carrier + x_drift, 'C0-', alpha=0.5, label="Signal")
+    >>> ax0.plot(t, x_drift, 'C2--', alpha=0.25, label="Drift")
+    >>> ax0.plot(t_out, x_res+x_env, 'C1.-', alpha=0.5, label="Envelope")
+    >>> ax0.plot(t_out, x_res-x_env, 'C1.-', alpha=0.5, label=None)
+    >>> ax0.grid(True)
+    >>> ax0.legend()
+    >>> plt.show()
+
+
+    The second example provides a geometric envelope interpretation of a complex-valued
+    signal: The following plot shows the complex-valued signal as a blue
+    3d-trajectory and the envelope as an orange round tube with varying diameter,
+    i.e., as :math:`|a(t)| \exp(j\rho)`, with :math:`\rho\in[-\pi,\pi]`. Also, the
+    projection into the 2d real and imaginary coordinate planes of trajectory and tube
+    is depicted.
+
+    The left plot shows an analytical signal, i.e, the phase difference between
+    real and imaginary is always 90 degrees resulting in a spiraling trajectory. It can
+    be seen that in this case the real part has also the expected envelope, i.e.,
+    representing the absolute value of the instantaneous amplitude. The right plot
+    shows a real-valued signal interpreted as a complex signal, i.e., having zero
+    imaginary part. There the resulting envelope is not as smooth as in the analytic
+    case and the instantaneous amplitude in the real plane is not recovered.
```

Comment:
I would divide this second example into two. 

The first example can be about the geometric envelope interpretation of a complex-valued signal. You would keep only the plot on the left for this one.

The second example must emphasize that `z(t) = a(t)` and `z(t) = a(t) + 0j` are interpreted differently and will yield different results. You could also possibly combine this example with the example about the real-valued signal you provide earlier.

Reply:
I expanded the description in commit bf9978fe201 to explicitly discuss the difference between a real-valued signal and the same signal with zero imaginary part, which also illustrates the relationship between those two plots. Hence, I think that separating those two plots would not bring any additional clarity npw.